parquet/src/main/java/org/apache/iceberg/parquet/ParquetValueReaders.java

```diff
@@ -189,6 +191,16 @@ public static ParquetValueReader<Long> int96Timestamps(ColumnDescriptor desc) {
     return new TimestampInt96Reader(desc);
   }
 
+  @SuppressWarnings("unchecked")
+  public static <T extends StructLike> ParquetValueReader<T> structLikeReader(
+      List<ParquetValueReader<?>> readers, Types.StructType struct, Class<T> structClass) {
+    if (structClass.equals(Record.class)) {
```

Comment:
`Record` extends from `StructLike` type. can we remove the `RecordReader`?  Maybe the only reason for potentially small performance  difference with `GenericRecord.copy()` in the `newStructData` method. 

```
        // GenericRecord.copy() is more performant than GenericRecord.create(StructType) since
        // NAME_MAP_CACHE access is eliminated. Using copy here to gain performance.
```

Reply:
Yeah I thought about this as well, I didn't want to break that since someone had spent the time to add in the comment.